base/sparse/sparsevector.jl

```diff
@@ -688,30 +746,26 @@ function show(io::IOContext, x::AbstractSparseVector)
     nzind = nonzeroinds(x)
     nzval = nonzeros(x)
     xnnz = length(nzind)
-
+    if length(nzind) == 0
+        return show(io, MIME("text/plain"), x)
+    end
     limit::Bool = get(io, :limit, false)
     half_screen_rows = limit ? div(displaysize(io)[1] - 8, 2) : typemax(Int)
     pad = ndigits(n)
-    sep = "\n\t"
     if !haskey(io, :compact)
         io = IOContext(io, :compact => true)
     end
     for k = 1:length(nzind)
         if k < half_screen_rows || k > xnnz - half_screen_rows
             print(io, "  ", '[', rpad(nzind[k], pad), "]  =  ")
             Base.show(io, nzval[k])
```

Comment:
Something else I found:

```
julia> a = speye(3,3)
3×3 SparseMatrixCSC{Float64,Int64} with 3 stored entries:
  [1, 1]  =  1.0
  [2, 2]  =  1.0
  [3, 3]  =  1.0

julia> b = similar(a, Float64, Int64)
3×3 SparseMatrixCSC{Float64,Int64} with 3 stored entries:
  [1, 1]  =  6.92397e-310
  [2, 2]  =  6.92397e-310
  [3, 3]  =  6.92397e-310

julia> Base.SparseArrays.dropstored!(a, 1, 1)
3×3 SparseMatrixCSC{Float64,Int64} with 2 stored entries:
  [2, 2]  =  1.0
  [3, 3]  =  1.0

julia> b
3×3 SparseMatrixCSC{Float64,Int64} with 2 stored entries:
  [2, 2]  =  6.92397e-310
  [3, 3]  =  6.92397e-310
```

`a` and `b` alias their internal representation. Seems odd.